shiftcommerce/flex-ruby-gem
# Issue: Remove breadcrumbs feature (#124)

### Overview

Related GitHub issue: #124 

PR removes the BreadCrumbs feature. 

### QA

1.) Add branch to local F/E, eg. 

```ruby
gem "flex_commerce_api", git: "https://github.com/shiftcommerce/flex-ruby-gem.git", branch: "issue/124_remove_breadcrumbs_feature"
``` 

OR 

use - https://breadcrumbsqa-integration.dev.shiftuatpreview.com/

2.) Perform surface  F/E  test by following the QA steps 
- https://github.com/shiftcommerce/flex-platform/wiki/Full-system-QA
- ensure breadcrumbs still appear on  the PDP and PLP pages

3.) Ensure Breadcrumbs cannot be accessed via gem

```bash
$ rails c

----------------------------------------------------------------------------------------------

$ FlexCommerce::Breadcrumb.all
=> NameError: uninitialized constant FlexCommerce::Breadcrumb

$ FlexCommerce:: BreadcrumbItem.all
=> NameError: uninitialized constant FlexCommerce::BreadcrumbItem

----------------------------------------------------------------------------------------------

$ category = FlexCommerce::Category.includes("breadcrumbs,parent,template_definition").where(category_tree_id: 29).find(5266).first

$ category.breadcrumbs
=> nil

# Check that relationships do not contain `breadcrumbs`

$  category.relationships
=> #<JsonApiClient::Relationships::Relations:0x007fc5b0e16e78
 @attributes=
  {"template_definition"=>
    {"links"=>
      {"self"=>"/integration/v1/categories/5266/relationships/template_definition.json_api",
       "related"=>"/integration/v1/categories/5266/template_definition.json_api"},
     "data"=>nil},
   "template"=>
    {"links"=>{"self"=>"/integration/v1/categories/5266/relationships/template.json_api", "related"=>"/integration/v1/categories/5266/template.json_api"}},
   "parent"=>
    {"links"=>{"self"=>"/integration/v1/categories/5266/relationships/parent.json_api", "related"=>"/integration/v1/categories/5266/parent.json_api"},
     "data"=>nil},
   "children"=>
    {"links"=>{"self"=>"/integration/v1/categories/5266/relationships/children.json_api", "related"=>"/integration/v1/categories/5266/children.json_api"}},
   "products"=>
    {"links"=>{"self"=>"/integration/v1/categories/5266/relationships/products.json_api", "related"=>"/integration/v1/categories/5266/products.json_api"}},
   "cat_menu_1"=>
    {"links"=>
      {"self"=>"/integration/v1/categories/5266/relationships/cat_menu_1.json_api", "related"=>"/integration/v1/categories/5266/cat_menu_1.json_api"}},
   "cat_menu_2"=>
    {"links"=>
      {"self"=>"/integration/v1/categories/5266/relationships/cat_menu_2.json_api", "related"=>"/integration/v1/categories/5266/cat_menu_2.json_api"}},
   "cat_menu_3"=>
    {"links"=>
      {"self"=>"/integration/v1/categories/5266/relationships/cat_menu_3.json_api", "related"=>"/integration/v1/categories/5266/cat_menu_3.json_api"}},
   "cat_menu_4"=>
    {"links"=>
      {"self"=>"/integration/v1/categories/5266/relationships/cat_menu_4.json_api", "related"=>"/integration/v1/categories/5266/cat_menu_4.json_api"}},
   "cat_menu_5"=>
    {"links"=>
      {"self"=>"/integration/v1/categories/5266/relationships/cat_menu_5.json_api", "related"=>"/integration/v1/categories/5266/cat_menu_5.json_api"}},
   "cat_menu_6"=>
    {"links"=>
      {"self"=>"/integration/v1/categories/5266/relationships/cat_menu_6.json_api", "related"=>"/integration/v1/categories/5266/cat_menu_6.json_api"}}},
 @changed_attributes={}>

----------------------------------------------------------------------------------------------

$ product = FlexCommerce::Product.includes("breadcrumbs.breadcrumb_items.item,variants.asset_files,variants.product,asset_files.related_files,template_definition").find(45084).first

$ product.breadcrumbs
=> nil

# Check that relationships do not contain `breadcrumbs`

$  product.relationships
=> #<JsonApiClient::Relationships::Relations:0x007fc5ade280b8
 @attributes=
  {"template_definition"=>
    {"links"=>
      {"self"=>"/integration/v1/products/45084/relationships/template_definition.json_api",
       "related"=>"/integration/v1/products/45084/template_definition.json_api"},
     "data"=>nil},
   "template"=>
    {"links"=>{"self"=>"/integration/v1/products/45084/relationships/template.json_api", "related"=>"/integration/v1/products/45084/template.json_api"}},
   "variants"=>
    {"links"=>{"self"=>"/integration/v1/products/45084/relationships/variants.json_api", "related"=>"/integration/v1/products/45084/variants.json_api"},
     "data"=>[{"type"=>"variants", "id"=>"220812"}]},
   "bundles"=>
    {"links"=>{"self"=>"/integration/v1/products/45084/relationships/bundles.json_api", "related"=>"/integration/v1/products/45084/bundles.json_api"}},
   "asset_files"=>
    {"links"=>
      {"self"=>"/integration/v1/products/45084/relationships/asset_files.json_api", "related"=>"/integration/v1/products/45084/asset_files.json_api"},
     "data"=>[{"type"=>"asset_files", "id"=>"120635"}]}},
 @changed_attributes={}>
```
